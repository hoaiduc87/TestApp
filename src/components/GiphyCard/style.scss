@import 'assets/styles/mixins.scss';

.giphy-item {
  .giphy-card {
    padding: 16px;
    background-color: $white;
    box-shadow: 0px 0px 2px 2px $grey-1;
    border-radius: 4px;

    &__image {
      position: relative;
      width: 100%;
      padding-top: 100%;
      cursor: pointer;

      img {
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        object-fit: cover;
        width: 100%;
        height: 100%;
      }
    }

    &__footer {
      margin-top: 8px;
      display: flex;
      flex-wrap: wrap;
      justify-content: space-between;

      &--right {
        display: flex;
        flex-wrap: wrap;
        align-items: center;
      }
    }

    .analytics-attachment-icon {
      margin-right: 4px;
      width: 28px;
      transform: rotateZ(135deg);
    }

    .analytics-item {
      margin-right: 16px;
      display: inline-flex;
      align-items: center;
      color: $grey-2;
      font-weight: 700;

      &:last-child {
        margin-right: 0;
      }

      img {
        width: 18px;
        margin-right: 4px;
      }

      span {
        word-break: break-word;
      }
    }
  }

  .giphy-author {
    display: flex;
    align-items: center;
    padding: 16px;

    &__image {
      flex-shrink: 0;
      width: 26px;
      height: 30px;
      border-radius: 50%;
    }

    &__name {
      margin-left: 8px;
      color: #4992BF;
      font-size: 20px;
      font-weight: 700;
      word-break: break-word;
    }
  }

  @include tablet {
    .giphy-card {
      padding: 12px;

      &__footer {
        margin-top: 6px;
      }

      .analytics-attachment-icon {
        width: 20px;
      }

      .analytics-item {
        margin-right: 10px;
        font-size: 12px;

        img {
          width: 13px;
        }
      }
    }

    .giphy-author {
      padding: 12px;

      &__image {
        width: 22px;
        height: 26px;
      }

      &__name {
        font-size: 16px;
      }
    }
  }

  @include mobile {
    .giphy-card {
      padding: 8px;

      &__footer {
        margin-top: 4px;
      }

      .analytics-attachment-icon {
        width: 16px;
      }

      .analytics-item {
        margin-right: 8px;
        font-size: 10px;

        img {
          width: 10px;
        }
      }
    }

    .giphy-author {
      padding: 8px;

      &__image {
        width: 18px;
        height: 22px;
      }

      &__name {
        font-size: 12px;
      }
    }
  }
}
