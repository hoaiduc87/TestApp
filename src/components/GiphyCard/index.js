import React, { memo } from 'react';
import PropTypes from 'prop-types';

import AttachmentIcon from 'assets/images/attachment.svg';
import ChatBubbleIcon from 'assets/images/chat_bubble.svg';
import EyeIcon from 'assets/images/eye.svg';
import HeartIcon from 'assets/images/heart.svg';

import './style.scss';

function GiphyCard({
  data,
  onClickImage,
}) {
  const { images, user, analytics } = data;
  const originalImage = (images || {}).original || {};
  const { load, sent, click } = analytics || {};

  return (
    <div className="giphy-item">
      <div className="giphy-card">
        <div className="giphy-card__image" onClick={() => onClickImage(originalImage.url)}>
          <img src={originalImage.url} alt="thumb" />
        </div>
        <div className="giphy-card__footer">
          <div className="giphy-card__footer--left">
            <img className="analytics-attachment-icon" src={AttachmentIcon} alt="" />
          </div>
          <div className="giphy-card__footer--right">
            <span className="analytics-item">
              <img src={EyeIcon} alt="" />
              <span>{load || 0}</span>
            </span>
            <span className="analytics-item">
              <img src={ChatBubbleIcon} alt=""/>
              <span>{sent || 0}</span>
            </span>
            <span className="analytics-item">
              <img src={HeartIcon} alt=""/>
              <span>{click || 0}</span>
            </span>
          </div>
        </div>
      </div>
      {!!user &&
        <div className="giphy-author">
          <img className="giphy-author__image" src={user.avatar_url} alt="user" />
          <strong className="giphy-author__name">{user.display_name || user.username || ''}</strong>
        </div>
      }
    </div>
  )
}

GiphyCard.propTypes = {
  data: PropTypes.object,
  onClickImage: PropTypes.func,
};

export default memo(GiphyCard);
