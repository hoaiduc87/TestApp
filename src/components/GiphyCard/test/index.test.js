import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import GiphyCard from '..';

const props = {
  data: {
    images: {
      original: {
        url: 'test'
      }
    },
    user: {
      avatar_url: '',
      display_name: 'test'
    },
    analytics: {
      load: 100,
      sent: 100,
      click: 100
    }
  }
};

describe('<GiphyCard/>', () => {
  it('Should run and match snapshot', () => {
    const {
      container: { firstChild }
    } = render(<GiphyCard {...props} />);
    expect(firstChild).toMatchSnapshot();
  });

  it('Should execute onClickImage once click image', () => {
    const onClickImage = jest.fn();
    const comp = shallow(<GiphyCard {...props} onClickImage={onClickImage} />);
    comp.find('.giphy-card__image').first().simulate('click');
    expect(onClickImage).toBeCalledWith('test');
  })
})