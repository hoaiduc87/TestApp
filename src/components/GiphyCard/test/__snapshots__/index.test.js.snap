// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiphyCard/> Should run and match snapshot 1`] = `
<div
  class="giphy-item"
>
  <div
    class="giphy-card"
  >
    <div
      class="giphy-card__image"
    >
      <img
        alt="thumb"
        src="test"
      />
    </div>
    <div
      class="giphy-card__footer"
    >
      <div
        class="giphy-card__footer--left"
      >
        <img
          alt=""
          class="analytics-attachment-icon"
          src="attachment.svg"
        />
      </div>
      <div
        class="giphy-card__footer--right"
      >
        <span
          class="analytics-item"
        >
          <img
            alt=""
            src="eye.svg"
          />
          <span>
            100
          </span>
        </span>
        <span
          class="analytics-item"
        >
          <img
            alt=""
            src="chat_bubble.svg"
          />
          <span>
            100
          </span>
        </span>
        <span
          class="analytics-item"
        >
          <img
            alt=""
            src="heart.svg"
          />
          <span>
            100
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    class="giphy-author"
  >
    <img
      alt="user"
      class="giphy-author__image"
      src=""
    />
    <strong
      class="giphy-author__name"
    >
      test
    </strong>
  </div>
</div>
`;
